ethcore/src/client/client.rs

```diff
@@ -845,6 +845,11 @@ impl Client {
 		*self.exit_handler.lock() = Some(Box::new(f));
 	}
 
+	/// Returns the chain reference
+	pub fn chain(&self) -> &RwLock<Arc<BlockChain>> {
```

Comment:
It's a bad practice to expose `&RwLock`. There is no guarantee this will not deadlock